### DISCUSSION TITLE
Questions about mmpose.evaluation.metrics.CocoMetric

### DISCUSSION THREAD

**Discussion by WinnerMeat** [open-mmlab/mmpose#2938] [Q&A] [answered]
In the configuration file of YOLOx-Pose, we need to set the content of val_evaluator, in which val_evaluator.score_mode has three optional parameters: bbox, bbox_keypoint and bbox_rle, in which bbox means that the score of the bbox will be used as the prediction result.
Is it understood that when val_evaluator.score_mode=bbox, the detection accuracy obtained by the program is the accuracy about the bounding box; val_evaluator.score_mode=bbox_keypoint, the detection accuracy obtained by the program is the accuracy about the keypoint of the human body.
The example of val_evaluator in the configuration file is as follows：
val_evaluator = dict(
    type='mmpose.CocoMetric',
    ann_file='***.json',
    score_mode='bbox_keypoint',
    nms_mode='none',
    use_area=False,
    iou_type='keypoints_crowd',
    prefix='crowdpose'
)
If anyone knows, I would appreciate an answer as soon as possible, thanks！

**Comment by Ben-Louis** [comment, marked as answer]
This parameter specifies how the confidence score of each prediction is calculated. For more information, you can refer to https://github.com/open-mmlab/mmpose/blob/5a3be9451bdfdad2053a90dc1199e3ff1ea1a409/mmpose/evaluation/metrics/coco_metric.py#L453-L475. In CocoMetric, the confidence score affects the order of predictions and, as a result, influences AP.

**Reply by WinnerMeat** [reply]
Thanks for the reply.
